apis/nucleus/src/components/Sheet.jsx

```diff
@@ -47,115 +51,141 @@ function getBounds(pos, columns, rows) {
   };
 }
 
-function Sheet({ model, halo, initialSnOptions, initialSnPlugins, initialError, onMount }) {
-  const { root } = halo;
-  const [layout] = useLayout(model);
-  const { theme: themeName, modelStore } = useContext(InstanceContext);
-  const [cells, setCells] = useState([]);
-  const [bgColor, setBgColor] = useState(undefined);
-  const [bgImage, setBgImage] = useState(undefined);
-  const [deepHash, setDeepHash] = useState('');
-
-  /// For each object
-  useEffect(() => {
-    if (layout) {
-      const hash = JSON.stringify(layout.cells);
-      if (hash === deepHash) {
-        return;
-      }
-      setDeepHash(hash);
-      const fetchObjects = async () => {
-        /*
+const Sheet = forwardRef(
+  (
+    { model: inputModel, halo, initialSnOptions, initialSnPlugins, initialError, onMount, unmount, navigation },
+    ref
+  ) => {
+    const { root } = halo;
+    const [model, setModel] = useState(inputModel);
+    const [layout] = useLayout(model);
+    const { theme: themeName, modelStore } = useContext(InstanceContext);
+    const [cells, setCells] = useState([]);
+    const [bgColor, setBgColor] = useState(undefined);
+    const [bgImage, setBgImage] = useState(undefined);
+    const [deepHash, setDeepHash] = useState('');
+    navigation?.setCurrentSheetId?.(model.id);
```

Comment:
As we want some kind of event off this we need to make sure it doesn't trigger for each render.

Reply:
There is a check if sheetId is changed or not in setCurrentSheetId, so it will do nothing if the sheetId is not changed.